strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,235 @@
+
+
+web_green1 = '#84ba5b'
+web_green_light1 = '#e0edd6'
+web_green2 = '#3e9651'
+web_red1 = '#d35e60'
+web_red2 = '#cc2529'
+sf_cyan = '#3e9bb7'
+sf_cyan_light = '#6eb4c9'
+sf_lightgreen = '#daf0e9'
+sf_red = '#d35e60'
+sf_grey = '#efefef'
+light_grey = '#CDCDCD'
+
+graph_node_colour = web_green2
+graph_edge_colour = light_grey
+subgraph_node_colour = web_red2
+subgraph_edge_colour = web_red2
+
+graph_node_size = 14
+subgraph_node_size = 16
+
+
+def plot(graph: nx.Graph, subgraph: Optional[list] = None, size: float = 500) -> None:
+    """ Creates a plot.ly plot of the input graph.
+
+    The graph layout is fixed to be the Kamada-Kawai layout with an aspect ratio of 1:1. The
+    function can plot just the input graph or the graph with a specified subgraph highlighted.
+    The function uses the standard colour theme of green nodes, grey edges, and red highlighted
+    subgraph.
+
+    **Example usage**:
+        >>> graph = nx.complete_graph(10)
+        >>> fig = plot(graph, [0, 1, 2, 3])
+        >>> fig.show()
+
+        Args:
+            graph (nx.Graph): input graph
+            subgraph (list): list of nodes comprising the subgraph to highlight
+            size (dict): size of the plot
+
+        Returns:
+             plot.ly graph object
+        """
+
+    s = graph.subgraph(subgraph)
+    l = nx.kamada_kawai_layout(graph)
+
+    g_nodes = go.Scatter(
+        **_node_coords(graph, l),
+        mode='markers',
+        hoverinfo='text',
+        marker=dict(color=graph_node_colour, size=graph_node_size, line_width=2)
+    )
+
+    g_edges = go.Scatter(
+        **edge_coords(graph, l),
+        line=dict(width=1, color=graph_edge_colour),
+        hoverinfo='none',
+        mode='lines'
+    )
+
+    g_nodes.text = [str(i) for i in graph.nodes()]
+
+    layout = go.Layout(showlegend=False, hovermode='closest',
+                       xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       yaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       margin=dict(b=0, l=0, r=0, t=25),
+                       height=size,
+                       width=size,
+                       plot_bgcolor='#ffffff'
+                       )
+
+    if subgraph:
+
+        s_edges = go.Scatter(
+            **edge_coords(s, l),
+            line=dict(width=2, color=subgraph_edge_colour),
+            hoverinfo='none',
+            mode='lines'
+        )
+
+        s_nodes = go.Scatter(
+            **_node_coords(s, l),
+            mode='markers',
+            hoverinfo='text',
+            marker=dict(color=subgraph_node_colour, size=subgraph_node_size, line_width=2)
+        )
+
+        s_nodes.text = [str(i) for i in s.nodes()]
+
+        f = go.Figure(data=[g_edges, s_edges, g_nodes, s_nodes], layout=layout)
+
+    else:
+        f = go.Figure(data=[g_edges, g_nodes], layout=layout)
+
+    return f
+
+
+def plot_subgraph(subgraph: nx.Graph, size: float = 500) -> None:
```

Comment:
I think we have artistic disagreement on this bit 😆 But for me it's really cool if we can plot the subgraph in the same layout as it would appear in the original graph. Could this be an option (can default to false)?

Reply:
I don't like plotting the subgraph in the same layout for two main reasons:
1. The subgraph already stands out a lot when highlighted in the original graph, so effectively removing the background graph doesn't much change things.
2. This is the most important reason: the subgraph layout within the graph is often doing very little to visually provide information about the subgraph. When we plot the subgraph on its own accord, it is much easier to analyze it. Consider the example below, it's much more obvious to see that there is a connecting path when we plot the subgraph with its own layout.
![newplot (4)](https://user-images.githubusercontent.com/40441298/65801411-10112f00-e147-11e9-9cf9-3af8edd64af6.png)
![newplot (5)](https://user-images.githubusercontent.com/40441298/65801417-156e7980-e147-11e9-866d-ddaf5aedaf89.png)